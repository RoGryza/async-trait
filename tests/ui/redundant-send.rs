use async_trait::async_trait;

#[async_trait(?Send)]
pub trait Trait {
    #[async_trait(?Send)]
    async fn method();
}

pub struct Struct;

#[async_trait(?Send)]
impl Trait for Struct {
    #[async_trait(?Send)]
    async fn method() {}
}

fn main() {}
