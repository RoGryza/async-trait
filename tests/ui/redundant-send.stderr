error: redundant #[async_trait(?Send)] in method
 --> tests/ui/redundant-send.rs:5:5
  |
5 |     #[async_trait(?Send)]
  |     ^^^^^^^^^^^^^^^^^^^^^

error: redundant #[async_trait(?Send)] in method
  --> tests/ui/redundant-send.rs:13:5
   |
13 |     #[async_trait(?Send)]
   |     ^^^^^^^^^^^^^^^^^^^^^
